src/main/java/com/example/demo/Service/EmailService.java

```diff
@@ -0,0 +1,83 @@
+package com.example.demo.Service;
+
+
+import com.example.demo.Model.Email;
+import com.example.demo.Repository.EmailRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class EmailService {
+
+    private EmailRepository emailRepository;
+
+    public EmailService(EmailRepository emailRepository) {
+        this.emailRepository = emailRepository;
+    }
+
+    public List<Email> getUnsentEmails(){
+        //I'm not sure if i should invoke method from repository, or it's 'wrapper' from this service.
+        return emailRepository.readEmailsFromFile().stream()
+                .filter(el -> !el.getSend())
+                .collect(Collectors.toList());
+    }
+
+    public List<Email> getSentEmails(){
+        //Same here.
+        return emailRepository.readEmailsFromFile().stream()
+                .filter(Email::getSend)
+                .collect(Collectors.toList());
+    }
+
+    public List<Email> getAllEmails(){
+        return emailRepository.readEmailsFromFile();
+    }
+
+    public List<Email> getEmailsToSend(){
+        //Returns emails that haven' been sent earlier.
+        //In order to know if the date has passed or not i compare email's getDate with current date,
+        //... by casting both of them to milliseconds.
+        Date currentDate = new Date();
+        return getUnsentEmails().stream()//Re-using code from above.
+                .filter(el -> el.getDate().getTime() < currentDate.getTime())
+                .collect(Collectors.toList());
+    }
+
+    public void add(Email email){
+        List<Email> presentEmails = emailRepository.readEmailsFromFile();
+
+        //I used Arrays.asList method in emailRepository, so the list it returns is immutable.
+        //In order to modify it i have to create new list, and then add all elements from the first list to it.
+        //It's a kostyl', isn't it? How do i bypass it?
```

Comment:
you need to create method to add just one email in you repository. You don't need to return an internal list outside of it. Instead of returning everything you have in the file you can created dedicated methods which will do the filtering. Repository is the right place for this